src/LowerTriangularMatrices/lower_triangular_matrix.jl

```diff
@@ -339,17 +342,25 @@ end
 
 Base.:(*)(s::Number, L::LowerTriangularArray) = L*s         # commutative
 Base.:(/)(L::LowerTriangularArray, s::Number) = L*inv(s)
-Base.:(/)(s::Number, L::LowerTriangularArray) = L/s
 
-# TODO: what if m / n unequal, do a boundscheck? 
-Base.:(+)(L1::LowerTriangularArray{T,N,ArrayType}, L2::LowerTriangularArray{T,N,ArrayType}) where {T,N,ArrayType} = LowerTriangularArray{T,N,ArrayType}(L1.data + L2.data, L1.m, L1.n)
+function Base.:(+)(L1::LowerTriangularArray{T,N,ArrayType}, L2::LowerTriangularArray{T,N,ArrayType}) where {T,N,ArrayType} 
+    @boundscheck (L1.m == L2.m) && (L1.n == L2.n) || throw(BoundsError)
+    LowerTriangularArray{T,N,ArrayType}(L1.data + L2.data, L1.m, L1.n)
+end
+
 function Base.:(+)(L1::LowerTriangularArray{T,N,ArrayType}, L2::LowerTriangularArray{S,N,ArrayType}) where {T,S,N,ArrayType}
+    @boundscheck (L1.m == L2.m) && (L1.n == L2.n) || throw(BoundsError)
     R = promote_type(T, S)
     LowerTriangularArray{R,N,ArrayType}(R.(L1.data) + R.(L2.data), L1.m, L1.n)
```

Comment:
Wouldn't the type promotion be handled by `L1.data + L2.data` anyway?

Reply:
so like do `L1.data + L2.data` in one like and then convert it to a LowerTriangularArray with the eltype of the result?